### PR TITLE
[1936] Switch to Cypress 12.11.0

### LINKED ISSUE #1936: Switch to cypress 12.10.0
* [x] **I have checked that this feature has not yet been suggested by someone else.**

Actual used cypress version (9.1.1), has a CVE https://devhub.checkmarx.com/cve-details/Cx10578cb2-c0fc/.
It has been corrected in versions higher to 10.3.0.

### PR DESCRIPTION
**Pull request by frouene**
Bug: https://github.com/eclipse-sirius/sirius-components/issues/1936

# Pull request template

## General purpose
What is the main goal of this pull request?
- [ ] Bug fixes
- [ ] New features
- [ ] Documentation
- [ ] Cleanup
- [ ] Tests
- [x] Build / releng

## Project management
- [ ] Has the pull request been added to the relevant project and milestone? (Only if you know that your work is part of a specific iteration such as the current one)
- [ ] Have the `priority:` and `pr:` labels been added to the pull request? (In case of doubt, start with the labels `priority: low` and `pr: to review later`)
- [ ] Have the relevant issues been added to the pull request?
- [ ] Have the relevant labels been added to the issues? (`area:`, `difficulty:`, `type:`)
- [ ] Have the relevant issues been added to the same project and milestone as the pull request?
- [ ] Has the `CHANGELOG.adoc` been updated to reference the relevant issues?
- [ ] Have the relevant API breaks been described in the `CHANGELOG.adoc`? (Including changes in the GraphQL API)
- [ ] In case of a change with a visual impact, are there any screenshots in the `CHANGELOG.adoc`? For example in `doc/screenshots/2022.5.0-my-new-feature.png`

## Architectural decision records (ADR)
- [ ] Does the title of the commit contributing the ADR start with `[doc]`?
- [ ] Are the ADRs mentioned in the relevant section of the  `CHANGELOG.adoc`?

## Dependencies
- [ ] Are the new / upgraded dependencies mentioned in the relevant section of the `CHANGELOG.adoc`?
- [ ] Are the new dependencies justified in the `CHANGELOG.adoc`?

## Frontend

This section is not relevant if your contribution does not come with changes to the frontend.

### General purpose
- [ ] Is the code properly tested? (Plain old JavaScript tests for business code and tests based on React Testing Library for the components)

### Typing
We need to improve the typing of our code, as such, we require every contribution to come with proper TypeScript typing for both changes contributing new files and those modifying existing files.
Please ensure that the following statements are true for each file created or modified (this may require you to improve code outside of your contribution).

- [ ] Variables have a proper type
- [ ] Functions’ arguments have a proper type
- [ ] Functions’ return type are specified
- Hooks are properly typed:
	- [ ] `useMutation<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useQuery<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useSubscription<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useMachine<CONTEXT_TYPE, EVENTS_TYPE>(…)`
	- [ ] `useState<STATE_TYPE>(…)`
- [ ] All components have a proper typing for their props
- [ ] No useless optional chaining with `?.` (if the GraphQL API specifies that a field cannot be `null`, do not treat it has potentially `null` for example)
- [ ] Nullable values have a proper type (for example `let diagram: Diagram | null = null;`)

## Backend

This section is not relevant if your contribution does not come with changes to the backend.

### General purpose
- [ ] Are all the event handlers tested?
- [ ] Are the event processor tested?
- [ ] Is the business code (services) tested?
- [ ] Are diagram layout changes tested?

### Architecture
- [ ] Are data structure classes properly separated from behavioral classes?
- [ ] Are all the relevant fields final?
- [ ] Is any data structure mutable? If so, please write a comment indicating why
- [ ] Are behavioral classes either stateless or side effect free?

### How to test this PR?
_Please describe here the various use cases to test this pull request_

- [ ] Has the Kiwi TCMS test suite been updated with tests for this contribution?

### CODE REVIEW

File: CHANGELOG.adoc
Comment (pcdavid) [inline CHANGELOG.adoc]: We will wait after 2023.4.0 is released to merge this PR, so before merging this will need to be moved into the (to be created) "2023.6.0 (unreleased)" section of the CHANGELOG.

**Review by pcdavid — DISMISSED** [review]
No issue with the actual content of the PR, the "requested change" is just to make sure we do not merge yet and don't forget to move the changelog entry to 2023.6.0.

**Review by pcdavid — COMMENTED** [review]
The upgrade seems to work fine, but I'm not a fan of the changes in the new version of Cypress. Especially the fact that it does not seem possible anymore to launch the whole test suite manually.

It seems that [it's possible to re-enable this](https://docs.cypress.io/guides/references/experiments#End-to-End-Testing) with a configuration  flag (`experimentalRunAllSpecs: true`) but I have not tested this.

**Comment by pcdavid** [comment]
Also: there's been [a 12.11.0 version](https://docs.cypress.io/guides/references/changelog?ref=cypress.io#12-11-0) released since. As long as we upgrade we should move to the very latest. From the changelog it should not require any change except the version number.

**Comment by frouene** [comment]
I switched to the latest version, i.e. 12.11.0

File: CHANGELOG.adoc
Comment (pcdavid) [inline CHANGELOG.adoc]: 12.11.0 now

**Comment by frouene** [comment]
> The upgrade seems to work fine, but I'm not a fan of the changes in the new version of Cypress. Especially the fact that it does not seem possible anymore to launch the whole test suite manually.
> 
> It seems that [it's possible to re-enable this](https://docs.cypress.io/guides/references/experiments#End-to-End-Testing) with a configuration flag (`experimentalRunAllSpecs: true`) but I have not tested this.

The ability to run all tests has been removed in version 10.0 because the feature doesn't work correctly (https://github.com/cypress-io/cypress/discussions/21628). Since version 12, it's possible to re-activate it but there is still problems with it.

File: CHANGELOG.adoc
Comment (frouene) [reply CHANGELOG.adoc]: Changelog corrected with the new version

**Review by sbegaudeau — APPROVED** [review]

**Review by sbegaudeau — APPROVED** [review]

**Pull request merged**
